src/handlers/setDependencyVersionHandler.ts

```diff
@@ -0,0 +1,198 @@
+        dependencyNodeInManagement = dependenciesNode.children?.find(node =>
+            node.children?.find(id => id.tag === XmlTagName.GroupId && id.text === gid) &&
+            node.children?.find(id => id.tag === XmlTagName.ArtifactId && id.text === aid)
+        );
+        targetText = constructDependencyNode(gid, aid, version, `${baseIndent}${indent}`, indent, eol);
+    } else if (targetNode.tag === XmlTagName.DependencyManagement && dependenciesNode === undefined) {
+        insertPosition = currentDocument.positionAt(targetNode.contentStart);
+        targetText = constructDependenciesNode(gid, aid, version, baseIndent, indent, eol);
+    } else if (targetNode.tag === XmlTagName.Project) {
+        insertPosition = currentDocument.positionAt(targetNode.contentEnd);
+        targetText = constructDepedencyManagementNode(gid, aid, version, baseIndent, indent, eol);
+    } else {
+        return;
+    }
+
+    const edit: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
+    if (deleteNode) { // the version of ${gid:aid} dependency node already imported should be deleted
+        const versionItem: ElementNode | undefined = deleteNode.children?.find(node => node.tag === XmlTagName.Version);
+        if (versionItem) {
+            if (versionItem.contentStart === undefined || versionItem.contentEnd === undefined) {
+                throw new UserError("Invalid target XML node to delete.");
+            }
+            const versionString: string = "<version>";
+            const start: number = versionItem.contentStart - versionString.length;
+            const end: number = versionItem.contentEnd + versionString.length + 1;
+            const range = new vscode.Range(currentDocument.positionAt(start), currentDocument.positionAt(end));
+            edit.delete(currentDocument.uri, range);
+        }
+    }
+    if (dependencyNodeInManagement) { // ${gid:aid} dependency node that already exists in <dependencyManagement> shoule be deleted
+        if (dependencyNodeInManagement.contentStart === undefined || dependencyNodeInManagement.contentEnd === undefined) {
+            throw new UserError("Invalid target XML node to delete.");
+        }
+        const dependencyString: string = "<dependency>";
+        const start = dependencyNodeInManagement.contentStart - dependencyString.length;
+        const end = dependencyNodeInManagement.contentEnd + dependencyString.length + 1;
+        const range = new vscode.Range(currentDocument.positionAt(start), currentDocument.positionAt(end));
+        edit.delete(currentDocument.uri, range);
+    }
+
+    edit.insert(currentDocument.uri, insertPosition, targetText);
+    await vscode.workspace.applyEdit(edit);
+    const endingPosition: vscode.Position = currentDocument.positionAt(currentDocument.offsetAt(insertPosition) + targetText.length);
+    textEditor.revealRange(new vscode.Range(insertPosition, endingPosition));
+    vscode.workspace.saveAll();
```

Comment:
`vscode.workspace.saveAll();` is aggressive. What if you have other unsaved files that you don't want to save for the moment?

Reply:
try textEditor.document.save